sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusListener.cs

```diff
@@ -403,42 +403,43 @@ private async Task RunBatchReceiveLoopAsync(CancellationToken cancellationToken)
-                                    }
-
-                                    // Pass CancellationToken.None to allow autocompletion to finish even when shutting down
-                                    completeTasks.Add(receiver.CompleteMessageAsync(message, CancellationToken.None));
+                                    continue;
                                 }
 
-                                await Task.WhenAll(completeTasks).ConfigureAwait(false);
+                                // Pass CancellationToken.None to allow autocompletion to finish even when shutting down
+                                completeTasks.Add(receiver.CompleteMessageAsync(message, CancellationToken.None));
                             }
-                            else
+
+                            await Task.WhenAll(completeTasks).ConfigureAwait(false);
+                        }
+                        else if (!result.Succeeded)
+                        {
+                            // For failed executions, we abandon the messages regardless of the autoCompleteMessages configuration.
+                            // This matches the behavior that happens for single dispatch functions as the processor does the same thing
+                            // in the Service Bus SDK.
+
+                            List<Task> abandonTasks = new List<Task>();
+                            foreach (ServiceBusReceivedMessage message in messagesArray)
                             {
-                                List<Task> abandonTasks = new List<Task>();
-                                foreach (ServiceBusReceivedMessage message in messagesArray)
+                                // skip messages that were settled in the user's function
+                                if (input.MessageActions.SettledMessages.ContainsKey(message))
                                 {
-                                    // skip messages that were settled in the user's function
-                                    if (input.MessageActions.SettledMessages.ContainsKey(message))
-                                    {
-                                        continue;
-                                    }
-
-                                    // Pass CancellationToken.None to allow abandon to finish even when shutting down
-                                    abandonTasks.Add(receiver.AbandonMessageAsync(message, cancellationToken: CancellationToken.None));
+                                    continue;
                                 }
 
-                                await Task.WhenAll(abandonTasks).ConfigureAwait(false);
+                                // Pass CancellationToken.None to allow abandon to finish even when shutting down
+                                abandonTasks.Add(receiver.AbandonMessageAsync(message, cancellationToken: CancellationToken.None));
                             }
+
+                            await Task.WhenAll(abandonTasks).ConfigureAwait(false);
```

Comment:
In the off chance that one of the abandon calls throws and sessions are enabled, would this cause an issue because the block to release the session on L445 will get skipped?   Should we consider wrapping this in a `try/catch` ?

Reply:
It shouldn't matter because we end up closing the receiver anyway in the fall through catch block.